services/main-frontend/src/components/page-specific/manage/courses/id/chatbot/ChatbotConfigurationForm.tsx

```diff
@@ -1,27 +1,296 @@
+import { css } from "@emotion/css"
 import React from "react"
+import { useForm } from "react-hook-form"
 import { useTranslation } from "react-i18next"
 
+import { ChatbotConfiguration, NewChatbotConf } from "@/shared-module/common/bindings"
+import Accordion from "@/shared-module/common/components/Accordion"
 import Button from "@/shared-module/common/components/Button"
+import CheckBox from "@/shared-module/common/components/InputFields/CheckBox"
+import TextAreaField from "@/shared-module/common/components/InputFields/TextAreaField"
+import TextField from "@/shared-module/common/components/InputFields/TextField"
+import { respondToOrLarger } from "@/shared-module/common/styles/respond"
 
 interface Props {
-  closeEditor: () => void
+  onConfigureChatbot: (bot: NewChatbotConf) => void
+  onDeleteChatbot: (botId: string, botName: string) => void
+  oldChatbotConf: ChatbotConfiguration
 }
 
-const ChatbotConfigurationForm: React.FC<Props> = ({ closeEditor }) => {
+type ConfigureChatbotFields = Omit<NewChatbotConf, "course_id" | "maintain_azure_search_index">
+
+const itemCss = css`
+  flex: 1;
+  ${respondToOrLarger.sm} {
+    flex: 0 45%;
+  }
+`
+const textFieldCss = css`
+  width: auto;
+`
+
+const ChatbotConfigurationForm: React.FC<Props> = ({
+  onConfigureChatbot,
+  onDeleteChatbot,
+  oldChatbotConf,
+}) => {
   const { t } = useTranslation()
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    clearErrors,
+    setError,
+  } = useForm<ConfigureChatbotFields>({
+    defaultValues: {
+      chatbot_name: oldChatbotConf.chatbot_name,
+      enabled_to_students: oldChatbotConf.enabled_to_students,
+      prompt: oldChatbotConf.prompt,
+      initial_message: oldChatbotConf.initial_message,
+      weekly_tokens_per_user: oldChatbotConf.weekly_tokens_per_user,
+      daily_tokens_per_user: oldChatbotConf.daily_tokens_per_user,
+      temperature: oldChatbotConf.temperature,
+      top_p: oldChatbotConf.top_p,
+      frequency_penalty: oldChatbotConf.frequency_penalty,
+      presence_penalty: oldChatbotConf.presence_penalty,
+      response_max_tokens: oldChatbotConf.response_max_tokens,
+      use_azure_search: oldChatbotConf.use_azure_search,
+      hide_citations: oldChatbotConf.hide_citations,
+      use_semantic_reranking: oldChatbotConf.use_semantic_reranking,
+    },
+  })
+
+  const validateForm = (data: ConfigureChatbotFields): boolean => {
```

Comment:
I feel like this is not the idiomatic way to validate forms with react-hook-form. See this: https://react-hook-form.com/get-started#Applyvalidation